### ISSUE TITLE
Upgrade fails from 4.2 to 4.3

### ISSUE THREAD

**Issue by jasoncoposky** [irods/irods#3487]
It appears the files have made it to the file system but the upgrade logic does not appear to have been triggered.

**Comment by alanking** [comment]
I tried upgrading from 4.2.11 to 4.3.0 (commit 31d48a47a4942df688da94b30aa8a5b5210261bb ) and it worked fine for me. I know the upgrade logic is being triggered because the following message was displayed after my packages upgraded:
```
=======================================================================

Welcome to iRODS.

This upgrade of an iRODS server is currently incomplete.

iRODS uses rsyslog and logrotate for log file management. Your system
is missing configuration files for these dependencies.

Run the following command to complete the upgrade.

  sudo python3 /var/lib/irods/scripts/setup_rsyslog_and_logrotate.py

Once the script completes, you should have two new files at the
following locations.

  rsyslog  : /etc/rsyslog.d/00-irods.conf
  logrotate: /etc/logrotate.d/irods

All iRODS specific log files will be located under the following
directory.

  /var/log/irods

=======================================================================
```

Is there anything in particular I should be testing?

**Comment by korydraughn** [comment]
seems the best we can do is run the test suite following an upgrade. if that works, we're good.

how are existing plugins handled in this case? are they upgraded too? do the packages fail to install if plugin versions do not match the server?

these questions touch on #5752. if we can't answer these questions today, then we need to investigate and define what should happen in these cases.

**Comment by alanking** [comment]
I can test the plugins we are planning to support if desired. Is the idea to upgrade the server package and see if the installed plugins follow suit?

I think I will link to this issue as part of an effort to support upgrading in https://github.com/irods/irods_testing_environment

**Comment by trel** [comment]
Yes, we trust that our package dependencies are correct, and that the package manager will complain if there are no plugins that satisfy the installer's request.   In the case of the mtime plugin, the functionality was absorbed by the server.

**Comment by alanking** [comment]
So... we need to build - wait for it - yet another Test Matrix? It would cover the following scenarios:

- upgrading a package on which the server package depends
- upgrading the server package without explicitly upgrading packages on which it depends or which depend on it
- upgrading a package which depends on the server package without explicitly upgrading the server package

For each scenario, we need to see the versions for each level of package. The easiest set of packages to test against would be these, I think:
- irods-runtime
- irods-server
- irods-database-plugin-postgres

irods-database-plugin-postgres depends on irods-server which depends on irods-runtime.

The questions to ask in each upgrade scenario would be:
- Are we expecting the upgrade to succeed or fail?
- On success, what should be the resulting versions of each package after each upgrade?

Here are some questions that I have:
- Should we consider externals packages?
- What about patch versions for plugins (e.g. 4.3.0.2)?
- Will we be able to test this before the packages are available in the RENCI repositories? This is probably something that needs to be added to the testing environment: a local repository manager (e.g. https://manpages.ubuntu.com/manpages/xenial/man1/reprepro.1.html and https://wiki.centos.org/HowTos/CreateLocalRepos)
- Once everything is upgraded, what should we assert? Is running the test suite enough, or should we test some specific scenarios?

**Comment by alanking** [comment]
See also: https://github.com/irods/irods_testing_environment/issues/295
